dubbo-metadata-report/dubbo-metadata-report-redis/src/main/java/org/apache/dubbo/metadata/store/redis/RedisMetadataReport.java

```diff
@@ -16,28 +16,46 @@
  */
 package org.apache.dubbo.metadata.store.redis;
 
+import org.apache.dubbo.common.Constants;
 import org.apache.dubbo.common.URL;
 import org.apache.dubbo.common.logger.Logger;
 import org.apache.dubbo.common.logger.LoggerFactory;
 import org.apache.dubbo.metadata.identifier.MetadataIdentifier;
 import org.apache.dubbo.metadata.support.AbstractMetadataReport;
 import org.apache.dubbo.rpc.RpcException;
+
+import redis.clients.jedis.HostAndPort;
 import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisCluster;
 import redis.clients.jedis.JedisPool;
 import redis.clients.jedis.JedisPoolConfig;
 
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
 /**
  * RedisMetadataReport
  */
 public class RedisMetadataReport extends AbstractMetadataReport {
 
     private final static Logger logger = LoggerFactory.getLogger(RedisMetadataReport.class);
 
-    final JedisPool pool;
+    JedisPool pool;
+    Set<HostAndPort> jedisClusterNodes;
+
 
     public RedisMetadataReport(URL url) {
         super(url);
-        pool = new JedisPool(new JedisPoolConfig(), url.getHost(), url.getPort());
+        if (url.getParameter(Constants.CLUSTER_KEY, false)) {
```

Comment:
`CLUSTER_KEY` is also used for other purposes http://dubbo.apache.org/zh-cn/docs/user/demos/fault-tolerent-strategy.html.

You need to add a prefix for `CLUSTER_KEY` key in this module to avoid overridden with the same key from other modules.

Reply:
Sorry, I forgot the url passed in is generated from MetadataConfig.

Reply:
Ok, I just reuse this key in metadataConfig.